### PR TITLE
Add recipe for Orange3-Bioinformatics

### PR DESCRIPTION
**Pull request by JakaKokosar**

### CODE REVIEW

**Comment by JakaKokosar** [comment]
Depends on https://github.com/conda-forge/staged-recipes/pull/5183

**Comment by mbargull** [comment]
Given its subject-specificity, it probably makes sense to submit this to https://github.com/bioconda/bioconda-recipes instead.

**Comment by astaric** [comment]
@mbargull, this is an add-on for [Orange3](https://github.com/conda-forge/orange3-feedstock). Since the core program and [a bunch of other add-ons](https://github.com/search?utf8=✓&q=org%3Aconda-forge+orange3&type=) are already available on conda-forge, it makes sense to have this one on the same channel as well.

**Comment by mbargull** [comment]
Yes, I saw that, and, TBH, that made hesitate a bit to suggest it for Bioconda. But then again, there is always a question whether there is a >> 0 probability that the software might at some point depend on packages that are already in `bioconda`. We have `bioconda` --> `conda-forge` dependencies, but don't have (or want) `conda-forge` --> `bioconda` ones (+avoid duplicate packages on both channels).
And I only now saw https://github.com/conda-forge/staged-recipes/pull/5183 -- I don't know Genesis/Genialis but from a first glance that looks only bioinformatics-related. Hence, if that were introduced to Bioconda previously, we'd have the unwanted case I'm getting at.

**Comment by astaric** [comment]
There is also >>0 probability that some cython code will be added in the future. Is there a way to enable automatic builds for windows on bioconda?

It would also be nice to have this split of packages based on intended usage mentioned somewhere in the readme/docs. I have searched through both and I could not find bioconda mentioned anywhere.

**Comment by mbargull** [comment]
Oh, right, I tend to forget about Windows. Windows builds on Bioconda is something that is not going to happen any time soon (if ever) (due to human resource limitations). If Windows is one of the target platforms for `Orange3-Bioinformatics`, then it has to be on `conda-forge` for now.

**Comment by astaric** [comment]
orange3-bioinformatics should target all three plaforms (win, mac and linux).

BTW, is bioconda considering adopting the conda-forge structure/build scripts sometime in the future? I really like the feedstock ownership idea on conda-forge, but I saw that bioconda keeps only one repository with all the recipes.

**Comment by bgruening** [comment]
@astaric unlikely at this point, but if so there is no point in have a separate channel anymore I guess.

File: recipes/Orange3-Bioinformatics/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  number: 0
+  skip: True  # [py2k]
+  script: python setup.py install --single-version-externally-managed --record record.txt
+  preserve_egg_dir: True
+
+requirements:
+  build:
+    - python
+    - setuptools
+    - requests-cache
```
Comment (jakirkham) [inline recipes/Orange3-Bioinformatics/meta.yaml]: Is `requests-cache` needed at build time?

File: recipes/Orange3-Bioinformatics/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+about:
+  home: https://github.com/biolab/orange3-bioinformatics
+  license: GPL-3.0
+  license_family: GPL
+  license_file: LICENSE
+  summary: 'Orange3 add-on with common functionality for bioinformatics'
+
+  description: |
+    Orange Bioinformatics provides access to publicly available data, like GEO data sets, GO and KEGG. 
+    All features can be combined with powerful visualization, network exploration and data mining techniques from the Orange data mining framework.
```
Comment (jakirkham) [inline recipes/Orange3-Bioinformatics/meta.yaml]: Would try to wrap this text a bit more so it doesn't spill over.

File: recipes/Orange3-Bioinformatics/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  skip: True  # [py2k]
+  script: python setup.py install --single-version-externally-managed --record record.txt
```
Comment (jakirkham) [inline recipes/Orange3-Bioinformatics/meta.yaml]: We now [prefer `pip`]( https://conda-forge.org/docs/meta.html#use-pip ) for installs. Please update accordingly.

File: recipes/Orange3-Bioinformatics/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  preserve_egg_dir: True
+
+requirements:
+  build:
+    - python
+    - setuptools
```
Comment (jakirkham) [inline recipes/Orange3-Bioinformatics/meta.yaml]: Please replace `setuptools` with `pip` as per the comment above.

File: recipes/Orange3-Bioinformatics/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+  home: https://github.com/biolab/orange3-bioinformatics
+  license: GPL-3.0
+  license_family: GPL
+  license_file: LICENSE
+  summary: 'Orange3 add-on with common functionality for bioinformatics'
+
+  description: |
+    Orange Bioinformatics provides access to publicly available data, like GEO data sets, GO and KEGG. 
+    All features can be combined with powerful visualization, network exploration and data mining techniques from the Orange data mining framework.
+  doc_url: ""
```
Comment (jakirkham) [inline recipes/Orange3-Bioinformatics/meta.yaml]: Would drop if there is no specific URL to visit for docs.

File: recipes/Orange3-Bioinformatics/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+requirements:
+  build:
+    - python
+    - setuptools
+    - requests-cache
+  run:
+    - serverfiles
+    - requests
+    - requests-cache
+    - genesis-pyapi
```
Comment (jakirkham) [inline recipes/Orange3-Bioinformatics/meta.yaml]: Are there any version requirements that upstream specifies for these?

File: recipes/Orange3-Bioinformatics/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+  script: python setup.py install --single-version-externally-managed --record record.txt
+  preserve_egg_dir: True
+
+requirements:
+  build:
+    - python
+    - setuptools
+    - requests-cache
+  run:
+    - serverfiles
```
Comment (jakirkham) [inline recipes/Orange3-Bioinformatics/meta.yaml]: Should `orange3` be required? Not really sure how these work.

**Comment by jakirkham** [comment]
Looks good for the most part. Added a few comments, but none are particularly difficult. Also a few questions.

File: recipes/Orange3-Bioinformatics/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  skip: True  # [py2k]
+  script: python setup.py install --single-version-externally-managed --record record.txt
+  preserve_egg_dir: True
```
[the remark this answers is not in the document]
Comment (astaric) [reply recipes/Orange3-Bioinformatics/meta.yaml]: Orange uses setuptools entry points to discover add-ons.

This flag [has been suggested in the past](https://github.com/conda-forge/staged-recipes/pull/4737/files#r159994619) to make sure entry points get properly installer, though I have no idea what it does behind the scene.

File: recipes/Orange3-Bioinformatics/meta.yaml (hunk 6 above)
Comment (JakaKokosar) [reply recipes/Orange3-Bioinformatics/meta.yaml]: There are no versions specified in [requirements.](
https://github.com/biolab/orange3-bioinformatics/blob/master/requirements.txt)

**Comment by JakaKokosar** [comment]
@jakirkham I've made suggested changes. Thank you!

**Comment by jakirkham** [comment]
Thanks @JakaKokosar. :)

**Pull request merged**